tests/shadow/test_transforms.py

```diff
@@ -0,0 +1,440 @@
+        expected = qml.jacobian(lambda x: qml.math.real(exact_circuit(x)))(x)
+
+        assert qml.math.allclose(actual, expected, atol=1e-1)
+
+    @pytest.mark.jax
+    def test_backward_jax(self):
+        """Test the gradient of the state for the JAX interface"""
+        import jax
+        from jax import numpy as jnp
+
+        shadow_circuit = basic_entangler_circuit(3, shots=20000, interface="jax")
+        shadow_circuit = qml.shadows.state(wires=[0, 1, 2], diffable=True)(shadow_circuit)
+        exact_circuit = basic_entangler_circuit_exact_state(3, "jax")
+
+        # make rotations close to pi / 2 to ensure gradients are not too small
+        x = jnp.array(
+            np.random.uniform(
+                0.8, 2, size=qml.BasicEntanglerLayers.shape(n_layers=1, n_wires=3)
+            ).astype(np.complex64)
+        )
+
+        actual = qml.math.real(jax.jacrev(shadow_circuit, holomorphic=True)(x))
+        expected = qml.math.real(jax.jacrev(exact_circuit, holomorphic=True)(x))
+
+        assert qml.math.allclose(actual, expected, atol=1e-1)
+
+    @pytest.mark.tf
+    def test_backward_tf(self):
+        """Test the gradient of the state for the tensorflow interface"""
+        import tensorflow as tf
+
+        shadow_circuit = basic_entangler_circuit(3, shots=20000, interface="tf")
+        shadow_circuit = qml.shadows.state(wires=[0, 1, 2], diffable=True)(shadow_circuit)
+        exact_circuit = basic_entangler_circuit_exact_state(3, "tf")
+
+        # make rotations close to pi / 2 to ensure gradients are not too small
+        x = tf.Variable(
+            np.random.uniform(0.8, 2, size=qml.BasicEntanglerLayers.shape(n_layers=1, n_wires=3))
+        )
+
+        with tf.GradientTape() as tape:
+            out = shadow_circuit(x)
+
+        actual = tape.jacobian(out, x)
+
+        with tf.GradientTape() as tape2:
+            out2 = exact_circuit(x)
+
+        expected = tape2.jacobian(out2, x)
+
+        assert qml.math.allclose(actual, expected, atol=1e-1)
+
+    @pytest.mark.torch
+    def test_backward_torch(self):
+        """Test the gradient of the state for the torch interface"""
+        import torch
+
+        shadow_circuit = basic_entangler_circuit(3, shots=20000, interface="torch")
+        shadow_circuit = qml.shadows.state(wires=[0, 1, 2], diffable=True)(shadow_circuit)
+        exact_circuit = basic_entangler_circuit_exact_state(3, "torch")
+
+        # make rotations close to pi / 2 to ensure gradients are not too small
+        x = torch.tensor(
+            np.random.uniform(0.8, 2, size=qml.BasicEntanglerLayers.shape(n_layers=1, n_wires=3)),
+            requires_grad=True,
+        )
+
+        actual = torch.autograd.functional.jacobian(shadow_circuit, x)
+        expected = torch.autograd.functional.jacobian(exact_circuit, x)
+
+        assert qml.math.allclose(actual, expected, atol=1e-1)
+
+
+@pytest.mark.parametrize("diffable", [True, False])
+@pytest.mark.autograd
+class TestExpvalForward:
```

Comment:
Since this is tested already twice in shadow_expval and class tests, can we reduce these tests here to checking that the transform is applied correctly and maybe 1 integration test? Same for backward pass of the expvals.

Reply:
You're right. It's sufficient to check that the transform is correct, since we already test the forward/backward passes in the other tests.